core/server/master/src/main/java/alluxio/master/file/meta/TempInodePathForDescendant.java

```diff
@@ -87,7 +87,18 @@ public synchronized  InodeDirectory getParentInodeDirectory()
     if (mDescendantInode == null) {
       return super.getInodeList();
     }
-    throw new UnsupportedOperationException();
+    try {
+      if (super.getInode().getId() == mDescendantInode.getParentId()) {
+        // Only return the inode list if the descendant is the direct child.
```

Comment:
If `getInodeList` can detect when it is valid, then should `getParentInodeDirectory` also detect it?